sklearn/ensemble/tests/test_stacking_classifier.py

```diff
@@ -0,0 +1,217 @@
+    """Check parallel backend of StackingClassifier on iris."""
+    clf1 = LogisticRegression(random_state=123)
+    clf2 = RandomForestClassifier(random_state=123)
+    clf3 = GaussianNB()
+    clfm = GaussianNB()
+
+    eclf1 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm,
+        n_jobs=1).fit(X, y)
+    eclf2 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm,
+        n_jobs=2).fit(X, y)
+    assert_array_equal(eclf1.predict(X), eclf2.predict(X))
+    assert_array_equal(eclf1.predict_proba(X), eclf2.predict_proba(X))
+
+    # cv=1 option
+    eclf1 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm,
+        n_jobs=1, cv=1).fit(X, y)
+    eclf2 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm,
+        n_jobs=2, cv=1).fit(X, y)
+    assert_array_equal(eclf1.predict(X), eclf2.predict(X))
+    assert_array_equal(eclf1.predict_proba(X), eclf2.predict_proba(X))
+
+
+def test_sample_weight():
+    """Tests sample_weight parameter of StackingClassifier"""
+    decimal = 10
+
+    clf1 = LogisticRegression(random_state=123)
+    clf2 = RandomForestClassifier(random_state=123)
+    clf3 = GaussianNB()
+    clfm = GaussianNB()
+    eclf1 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm
+    ).fit(X, y, sample_weight=np.ones((len(y),)))
+    eclf2 = StackingClassifier(
+        estimators=[('lr', clf1), ('rf', clf2), ('nb', clf3)],
+        meta_estimator=clfm
+    ).fit(X, y)
+    assert_array_almost_equal(eclf1.predict(X), eclf2.predict(X),
+                              decimal=decimal)
+    assert_array_almost_equal(eclf1.predict_proba(X), eclf2.predict_proba(X),
+                              decimal=decimal)
+
+    sample_weight = np.random.RandomState(123).uniform(size=(len(y),))
+    clf4 = KNeighborsClassifier()
+    eclf3 = StackingClassifier(
+        estimators=[('lr', clf1), ('knn', clf4)],
+        meta_estimator=clfm)
+    msg = ("Underlying estimator knn of type "
+           "<class 'sklearn.neighbors.classification.KNeighborsClassifier'> "
+           "does not support `sample_weight`.")
+    assert_raise_message(ValueError, msg, eclf3.fit, X, y,
+                         sample_weight=sample_weight)
+    eclf4 = StackingClassifier(
+        estimators=[('lr', clf1), ('nb', clf3)],
+        meta_estimator=clf4)
+    msg = ("Underlying meta estimator "
+           "<class 'sklearn.neighbors.classification.KNeighborsClassifier'> "
+           "does not support `sample_weight`.")
+    assert_raise_message(ValueError, msg, eclf4.fit, X, y,
+                         sample_weight=sample_weight)
+
+
+def test_classify_iris():
+    """Check classification by majority label on dataset iris."""
```

Comment:
with nosetests, docstrings in test functions make test transcripts harder to read. make this a comment instead.